spec/models/firm_spec.rb

```diff
@@ -66,6 +66,37 @@
     end
   end
 
+  describe '#offices' do
+    let(:firm) { create(:firm) }
+    let!(:unsorted_offices) do
+      [
+        FactoryGirl.create(:office, firm: firm, address_line_one: 'd', created_at: Time.zone.now),
+        FactoryGirl.create(:office, firm: firm, address_line_one: 'b', created_at: 2.days.ago),
+        FactoryGirl.create(:office, firm: firm, address_line_one: 'a', created_at: 3.days.ago),
+        FactoryGirl.create(:office, firm: firm, address_line_one: 'c', created_at: 1.day.ago)
+      ]
+    end
+
+    describe 'default sort order' do
+      subject { firm.offices.map(&:address_line_one) }
+      it { is_expected.to eq(%w{a b c d}) }
+    end
+  end
+
+  describe '#main_office' do
+    let(:firm) { create(:firm) }
+    subject { firm.main_office }
+
+    context 'when the firm has no offices' do
+      it { is_expected.to be_nil }
+    end
+
+    context 'when the firm has offices' do
+      before { FactoryGirl.create_list(:office, 3, firm: firm) }
+      it { is_expected.to eq(firm.offices[0]).and(eq(firm.offices.first)) }
```

Comment:
Oh, do `firm.offices[0]` and `firm.offices.first` return different results from one another?